app/models/encounter.rb

```diff
@@ -27,6 +27,27 @@ def self.entity_scope
 
   attr_accessor :new_samples # Array({entity_id: String}) of new generated samples from UI.
 
+  # Finds an Encounter by its UUID or numerical ID.
+  def self.find_by_identifier(uuid_or_id)
+    if encounter = where(uuid: uuid_or_id).take
```

Comment:
We should be able to differentiate `uuid_or_id` based on the UUID format. That should be sufficient to issue only a single DB request either way.